cmd/log_index.cpp

```diff
@@ -0,0 +1,240 @@
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#include <iomanip>
+#include <iostream>
+#include <string>
+#include <unordered_map>
+
+#include <CLI/CLI.hpp>
+#include <boost/endian/conversion.hpp>
+#include <boost/filesystem.hpp>
+#include <thread>
+#include <silkworm/common/log.hpp>
+#include <silkworm/db/access_layer.hpp>
+#include <silkworm/db/bitmap.hpp>
+#include <silkworm/db/stages.hpp>
+#include <silkworm/db/tables.hpp>
+#include <silkworm/etl/collector.hpp>
+#include <cbor/decoder.h>
+
+using namespace silkworm;
+
+constexpr size_t kBitmapBufferSizeLimit = 512 * kMebi;
+
+
+void loader_function(etl::Entry entry, lmdb::Table *target_table, unsigned int db_flags) {
+    auto bm{roaring::Roaring::readSafe(byte_ptr_cast(entry.value.data()), entry.value.size())};
+    Bytes last_chunk_index(entry.key.size() + 4, '\0');
+    std::memcpy(&last_chunk_index[0], &entry.key[0], entry.key.size());
+    boost::endian::store_big_u32(&last_chunk_index[entry.key.size()], UINT32_MAX);
+    auto previous_bitmap_bytes{target_table->get(last_chunk_index)};
+    if (previous_bitmap_bytes.has_value()) {
+        bm |= roaring::Roaring::readSafe(byte_ptr_cast(previous_bitmap_bytes->data()),
+                                                previous_bitmap_bytes->size());
+        db_flags = 0;
+    }
+    while (bm.cardinality() > 0) {
+        auto current_chunk{db::bitmap::cut_left(bm, db::bitmap::kBitmapChunkLimit)};
+        // make chunk index
+        Bytes chunk_index(entry.key.size() + 4, '\0');
+        std::memcpy(&chunk_index[0], &entry.key[0], entry.key.size());
+        uint64_t suffix{bm.cardinality() == 0 ? UINT32_MAX : current_chunk.maximum()};
+        boost::endian::store_big_u32(&chunk_index[entry.key.size()], suffix);
+        Bytes current_chunk_bytes(current_chunk.getSizeInBytes(), '\0');
+        current_chunk.write(byte_ptr_cast(&current_chunk_bytes[0]));
+        target_table->put(chunk_index, current_chunk_bytes, db_flags);
+    }
+}
+
+class listener_log_index : public cbor::listener {
+    public:
+    listener_log_index(uint64_t block_number, std::unordered_map<std::string, roaring::Roaring> * topics_map,
+                std::unordered_map<std::string, roaring::Roaring> * addrs_map, uint64_t * allocated_topics, uint64_t * allocated_addrs_): 
+                block_number_(block_number), topics_map_(topics_map), addrs_map_(addrs_map), allocated_topics_(allocated_topics), allocated_addrs_(allocated_addrs_) {};
+
+    void on_integer(int){};
```

Comment:
All overridden methods should have `override` specifier, e.g.
```
void on_integer(int) override {}
```

Rationale: [Google's Style Guide](https://google.github.io/styleguide/cppguide.html#Inheritance).